arrow/arrow.py

```diff
@@ -1282,6 +1283,102 @@ def gather_timeframes(_delta: float, _frame: TimeFrameLiteral) -> float:
+            "seconds": 0,
+            "minutes": 0,
+            "hours": 0,
+            "days": 0,
+            "weeks": 0,
+            "months": 0,
+            "years": 0,
+        }
+
+        # Create a locale object based off given local
+        locale_obj = locales.get_locale(locale)
+
+        # Create a regex pattern object for numbers
+        num_pattern = re.compile(r"[0-9]+")
+
+        # Sign logic
+        sign_val = 0
+
+        future_string = locale_obj.future
+        future_string = future_string.format(".*")
+        future_pattern = re.compile(fr"{future_string}")
+        future_pattern_match = future_pattern.findall(timestring)
+
+        past_string = locale_obj.past
+        past_string = past_string.format(".*")
+        past_pattern = re.compile(fr"{past_string}")
+        past_pattern_match = past_pattern.findall(timestring)
+
+        if past_pattern_match:
+            sign_val = -1
+
+        elif future_pattern_match:
+            sign_val = 1
+
+        else:
+            raise ValueError(
+                "Invalid String. Needs to represent a future or past time length"
+            )
+
+        # Search timestring for each time unit within locale
+        for unit in locale_obj.timeframes:
+
+            # Numeric unit of change
+            change_value = 0
+
+            # Replace {0} with regex [0-9]
+            search_string = str(locale_obj.timeframes[unit])
+            search_string = search_string.format("[0-9]+")
+
+            # Create search pattern and find within string
+            pattern = re.compile(fr"{search_string}")
+            match = pattern.search(timestring)
+
+            # If there is no match continue to next iteration
+            if not match:
+                continue
+
+            match_string = match.group()
+            num_match = num_pattern.search(match_string)
+
+            # If no number matches set change value to be one
+            if not num_match:
+                change_value = 1
+
+            else:
+                change_value = int(num_match.group())
+
+            # No time to update if now is the unit
+            if unit == "now":
+                continue
+
+            # Add change value to the correct unit (incorporates the plualirty that exists within timeframe i.e second v.s seconds)
+            time_unit_to_change = str(unit)
+            time_unit_to_change += "s" if (str(time_unit_to_change)[-1] != "s") else ""
+            time_object_info[time_unit_to_change] = change_value
+
+        return current_time.shift(
+            seconds=sign_val * time_object_info["seconds"],
+            minutes=sign_val * time_object_info["minutes"],
+            hours=sign_val * time_object_info["hours"],
+            days=sign_val * time_object_info["days"],
+            weeks=sign_val * time_object_info["weeks"],
+            months=sign_val * time_object_info["months"],
+            years=sign_val * time_object_info["years"],
+        )
```

Comment:
```suggestion

        time_changes = {k: sign_val*v for k, v in time_object_info.items()}

        return current_time.shift(**time_changes)
```

Or something along those lines, saves having to do all the multiplications in the return.